src/commonControls/questionnaireControl/QuestionnaireControl.ts

```diff
@@ -0,0 +1,1578 @@
+            this.handleFunc = matches[0].handle.bind(this);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // tsDoc - see Control
+    async handle(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        const content = this.getQuestionnaireContent(input);
+        if (this.handleFunc === undefined) {
+            log.error(
+                'QuestionnaireControl: handle called but this.handlerFunc not set.  are canHandle/handle out of sync?',
+            );
+            throw new Error(`this.handlerFunc not set.  are canHandle/handle out of sync?`);
+        }
+
+        await this.handleFunc(input, resultBuilder);
+
+        this.state.activeInitiative = undefined; // clear the initiative state so we don't get confused on subsequent turns.
+
+        //NOTE: this is a trial of _not_ adding initiative during handle.. but rather
+        //letting the initiative phase always run.
+        // if (resultBuilder.hasInitiativeAct() !== true && (await this.canTakeInitiative(input)) === true) {
+        //     await this.takeInitiative(input, resultBuilder);
+        // }
+
+        const isDone = this.getFirstUnansweredQuestion(content) === undefined;
+        const confirmationRequired = this.evaluateBooleanProp(this.props.dialog.confirmationRequired, input);
+        if (isDone && !confirmationRequired) {
+            resultBuilder.addAct(new CompletedAct(this));
+        }
+    }
+
+    private isActivate(input: ControlInput): any {
+        try {
+            okIf(InputUtil.isIntent(input, GeneralControlIntent.name));
+            const { feedback, action, target } = unpackGeneralControlIntent(
+                (input.request as IntentRequest).intent,
+            );
+            okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, [$.Feedback.Affirm, $.Feedback.Disaffirm]));
+            okIf(InputUtil.actionIsMatch(action, this.props.interactionModel.actions.activate));
+            this.handleFunc = this.handleActivate;
+            return true;
+        } catch (e) {
+            verifyErrorIsGuardFailure(e);
+            return false;
+        }
+    }
+
+    // TODO: extract from each handle-method a programmatic method that can be public.
+    //  i.e. handling = extract data from input then call the business function.
+    //  this is a safer way to let folks create custom handlers that extract data from
+    //  different inputs.  the business functions are the 'functional capabilities'.
+    //  Much less risk of folks screwing up the state variables and breaking things.
+    //  We shouldn't let folks shoot themselves in the foot by forgetting to set/clear
+    //  flags etc.
+
+    private async handleActivate(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        this.setExplicitlyActivatedFlag();
```

Comment:
I think we had this discussion, maybe we need to `setExplicitlyActivatedFlag` on handle method so any interaction with controls triggers its activation

Reply:
hmm. yes I need to think more about this flag, the various ways the control can be activated and when it is ok to silently complete the questionnaire.  The answer to the latter is probably something like 'on first activation, all questions complete, confirmationRequired = false'.  

I think I will have to take this as a follow up as its a bit involved.

Reply:
I've removed the flag and tried a new approach but its still not quite right.  I will follow up.